tbl/screen.py

```diff
@@ -196,24 +225,39 @@ def cmd_input(screen, stdscr, prompt=""):
     """
     Prompts for and reads a line of input in the cmd line.
     """
-    translate = lambda c: {
-        127: 8,  # BACKSPACE -> C-h
-    }.get(c, c)
+
+    def _validate(c):
+        translate = {127: 8, # BACKSPACE -> C-h
+                     10: 7, # ENTER --> C-g
+                     343: 7 # RETURN --> C-g
+                     }
+
+        # Escape/C-g ==> abort, ala Emacs.
+        if c == 27 or c == 7:
```

Comment:
Err, I don't think emacs lets you abort the minibuffer by pressing escape.  This logic will prevent us from using meta combos while editing commands, and also will probably leave a spurious extra character available afters.  I suggest we only use C-g for this.